x-pack/plugins/lists/server/services/items/buffer_lines.ts

```diff
@@ -11,38 +11,99 @@ const BUFFER_SIZE = 100;
 
 export class BufferLines extends Readable {
   private set = new Set<string>();
+  private boundary: string | null = null;
+  private readableText: boolean = false;
+  private paused: boolean = false;
+
   constructor({ input }: { input: NodeJS.ReadableStream }) {
     super({ encoding: 'utf-8' });
+
     const readline = readLine.createInterface({
       input,
     });
 
+    // We are parsing multipart/form-data involving boundaries as fast as we can to get
+    // * The filename if it exists and emit it
+    // * The actual content within the multipart/form-data
     readline.on('line', (line) => {
-      this.push(line);
+      if (this.boundary == null && line.startsWith('--')) {
+        this.boundary = `${line}--`;
+      } else if (this.boundary != null && !this.readableText && line.trim() !== '') {
+        if (line.startsWith('Content-Disposition')) {
+          const fileNameMatch = RegExp('filename="(?<fileName>.+)"');
+          const matches = fileNameMatch.exec(line);
+          if (matches?.groups?.fileName != null) {
+            this.emit('fileName', matches.groups.fileName);
+          }
+        }
+      } else if (this.boundary != null && !this.readableText && line.trim() === '') {
+        // we are ready to be readable text now for parsing
+        this.readableText = true;
+      } else if (this.readableText && line.trim() === '') {
+        // skip and do nothing as this is either a empty line or an upcoming end is about to happen
+      } else if (this.boundary != null && this.readableText && line === this.boundary) {
+        // we are at the end of the stream
+        this.boundary = null;
+        this.readableText = false;
+      } else {
+        // we have actual content to push
+        this.push(line);
+      }
     });
 
     readline.on('close', () => {
       this.push(null);
     });
   }
 
-  public _read(): void {
-    // No operation but this is required to be implemented
+  public _read(): void {}
```

Comment:
I was gonna say we should leave the comment, but the typescript error generated if one removes this method should be straightforward 👍